kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWritesSuite.scala

```diff
@@ -849,6 +978,29 @@ class DeltaTableWritesSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBa
     }
   }
 
+  private def getMetadataActionFromCommit(
+    engine: Engine, table: Table, version: Long): Optional[Metadata] = {
+    val logPath = new Path(table.getPath(engine), "_delta_log")
+    val file = engine
+      .getFileSystemClient
+      .listFrom(FileNames.listingPrefix(logPath, version)).next
+    val columnarBatches =
+      engine.getJsonHandler.readJsonFiles(
+        singletonCloseableIterator(file),
+        SingleAction.FULL_SCHEMA,
+        Optional.empty())
+    if (!columnarBatches.hasNext) {
+      return null
+    }
+    val metadataVector = columnarBatches.next().getColumnVector(3)
+    Optional.ofNullable(Metadata.fromColumnVector(metadataVector, 1, engine))
```

Comment:
what happens if the metadata is not a rowId 1? I think we may want to iterate through the actions and find the first non-null `Metadata` object or null if we exhausted the iterator.

Reply:
Yes.